neqo-http3/src/request_stream_client.rs

```diff
@@ -255,6 +256,10 @@ impl RequestStreamClient {
                 RequestStreamClientState::BlockedDecodingHeaders { .. } => break Ok(()),
                 RequestStreamClientState::WaitingForData => {
                     self.recv_frame(conn)?;
+                    if self.state == RequestStreamClientState::ClosePending {
+                        // Received 0 byte fin? Client must see this.
+                        self.conn_events.borrow_mut().data_readable(self.stream_id);
```

Comment:
Do we need a self.add_event(blah) method rather than having this borrow_mut() thing scattered around?  It's not as though the constructor for the event is complicated.